Converting time series from GUI to SQL can cause frontend reload in some cases

**Describe the bug**
When converting a time series GUI question to SQL, then depending on the visualization type, can cause a frontend reload, which makes it difficult to convert question to SQL in some circumstances.
Regression since 0.42.0

**Workaround** To prevent the frontend reload, then change the visualization to Line before converting to SQL. It can be changed to Table after converted to SQL and the query has been executed once.

**To Reproduce**
1. Question > Sample > Orders - summarize Count group by CreatedAt:Month - visualize as Line, save question
![image](https://user-images.githubusercontent.com/1447303/162952483-3db3368e-2a9a-4ea5-b823-fa6dfea5a337.png)
2. Go to Notebook > View the SQL > Convert - notice it's showing the selector, which has no effect on SQL, so it should be hidden
![image](https://user-images.githubusercontent.com/1447303/162952864-9d1896c7-9563-4076-81fd-5f804249632c.png)
3. Changing visualization type to to Line will cause frontend reload. If the query had been executed before changing the visualization would not cause frontend reload.
4. Go back to the saved question from step 1. Change the visualize to Table - then go to Notebook > View the SQL > Convert - causes frontend reload

https://user-images.githubusercontent.com/1447303/162959212-204c320a-8999-4cc3-ab10-eb4a92c08417.mp4

<details><summary>Browser console</summary>

```
react-dom.production.min.js:209 TypeError: this._query.aggregations is not a function
    at n.value (Dimension.ts:1265:39)
    at n.value (Dimension.ts:1282:30)
    at fu (DimensionInfoPopover.tsx:37:49)
    at Qo (react-dom.production.min.js:153:146)
    at Ha (react-dom.production.min.js:175:309)
    at xs (react-dom.production.min.js:263:406)
    at yl (react-dom.production.min.js:246:265)
    at ml (react-dom.production.min.js:246:194)
    at ll (react-dom.production.min.js:239:172)
    at react-dom.production.min.js:123:115
os @ react-dom.production.min.js:209
Ms.o.componentDidCatch.n.callback @ react-dom.production.min.js:227
vo @ react-dom.production.min.js:131
us @ react-dom.production.min.js:212
Ml @ react-dom.production.min.js:255
t.unstable_runWithPriority @ scheduler.production.min.js:19
Vi @ react-dom.production.min.js:122
wl @ react-dom.production.min.js:248
ll @ react-dom.production.min.js:239
(anonymous) @ react-dom.production.min.js:123
t.unstable_runWithPriority @ scheduler.production.min.js:19
Vi @ react-dom.production.min.js:122
Zi @ react-dom.production.min.js:123
Ki @ react-dom.production.min.js:122
rl @ react-dom.production.min.js:230
enqueueSetState @ react-dom.production.min.js:132
_.setState @ react.production.min.js:12
(anonymous) @ ExplicitSize.jsx:105
c @ throttle.js:32
(anonymous) @ ExplicitSize.jsx:64
Visualization.jsx:101 Error caught in <Visualization> TypeError: this._query.aggregations is not a function
    at n.value (Dimension.ts:1265:39)
    at n.value (Dimension.ts:1282:30)
    at fu (DimensionInfoPopover.tsx:37:49)
    at Qo (react-dom.production.min.js:153:146)
    at Ha (react-dom.production.min.js:175:309)
    at xs (react-dom.production.min.js:263:406)
    at yl (react-dom.production.min.js:246:265)
    at ml (react-dom.production.min.js:246:194)
    at ll (react-dom.production.min.js:239:172)
    at react-dom.production.min.js:123:115 {componentStack: '\n    in fu\n    in div\n    in DraggableCore\n    in …\n    in t\n    in DragDropContextProvider\n    in a'}
value @ Visualization.jsx:101
Ms.o.componentDidCatch.n.callback @ react-dom.production.min.js:227
vo @ react-dom.production.min.js:131
us @ react-dom.production.min.js:212
Ml @ react-dom.production.min.js:255
t.unstable_runWithPriority @ scheduler.production.min.js:19
Vi @ react-dom.production.min.js:122
wl @ react-dom.production.min.js:248
ll @ react-dom.production.min.js:239
(anonymous) @ react-dom.production.min.js:123
t.unstable_runWithPriority @ scheduler.production.min.js:19
Vi @ react-dom.production.min.js:122
Zi @ react-dom.production.min.js:123
Ki @ react-dom.production.min.js:122
rl @ react-dom.production.min.js:230
enqueueSetState @ react-dom.production.min.js:132
_.setState @ react.production.min.js:12
(anonymous) @ ExplicitSize.jsx:105
c @ throttle.js:32
(anonymous) @ ExplicitSize.jsx:64
react-dom.production.min.js:209 Error: Minified React error #31; visit https://reactjs.org/docs/error-decoder.html?invariant=31&args[]=Error%3A%20An%20error%20occurred%20displaying%20this%20visualization.&args[]= for the full message or use the non-minified dev environment for full errors and additional helpful warnings.
    at To (react-dom.production.min.js:139:47)
    at react-dom.production.min.js:149:278
    at za (react-dom.production.min.js:173:103)
    at xs (react-dom.production.min.js:265:156)
    at yl (react-dom.production.min.js:246:265)
    at ml (react-dom.production.min.js:246:194)
    at ll (react-dom.production.min.js:239:172)
    at react-dom.production.min.js:123:115
    at t.unstable_runWithPriority (scheduler.production.min.js:19:467)
    at Vi (react-dom.production.min.js:122:325)
os @ react-dom.production.min.js:209
Ms.o.componentDidCatch.n.callback @ react-dom.production.min.js:227
vo @ react-dom.production.min.js:131
us @ react-dom.production.min.js:212
Ml @ react-dom.production.min.js:255
t.unstable_runWithPriority @ scheduler.production.min.js:19
Vi @ react-dom.production.min.js:122
wl @ react-dom.production.min.js:248
ll @ react-dom.production.min.js:239
(anonymous) @ react-dom.production.min.js:123
t.unstable_runWithPriority @ scheduler.production.min.js:19
Vi @ react-dom.production.min.js:122
Zi @ react-dom.production.min.js:123
Ki @ react-dom.production.min.js:122
rl @ react-dom.production.min.js:230
enqueueSetState @ react-dom.production.min.js:132
_.setState @ react.production.min.js:12
(anonymous) @ ExplicitSize.jsx:105
c @ throttle.js:32
(anonymous) @ ExplicitSize.jsx:64
StructuredQuery.ts:340 Warning: can't clean query without metadata!
value @ StructuredQuery.ts:340
value @ Question.ts:1210
value @ Question.ts:936
C @ urls.js:95
value @ Navbar.jsx:171
value @ Navbar.jsx:286
Wa @ react-dom.production.min.js:182
Ba @ react-dom.production.min.js:181
xs @ react-dom.production.min.js:263
yl @ react-dom.production.min.js:246
ml @ react-dom.production.min.js:246
ll @ react-dom.production.min.js:239
(anonymous) @ react-dom.production.min.js:123
t.unstable_runWithPriority @ scheduler.production.min.js:19
Vi @ react-dom.production.min.js:122
Zi @ react-dom.production.min.js:123
Ki @ react-dom.production.min.js:122
rl @ react-dom.production.min.js:230
enqueueSetState @ react-dom.production.min.js:132
_.setState @ react.production.min.js:12
(anonymous) @ ExplicitSize.jsx:105
c @ throttle.js:32
(anonymous) @ ExplicitSize.jsx:64
react-dom.production.min.js:209 TypeError: this._query.aggregations is not a function
    at n.value (Dimension.ts:1265:39)
    at n.value (Dimension.ts:1282:30)
    at fu (DimensionInfoPopover.tsx:37:49)
    at Qo (react-dom.production.min.js:153:146)
    at xs (react-dom.production.min.js:261:496)
    at yl (react-dom.production.min.js:246:265)
    at ml (react-dom.production.min.js:246:194)
    at ll (react-dom.production.min.js:239:172)
    at react-dom.production.min.js:123:115
    at t.unstable_runWithPriority (scheduler.production.min.js:19:467)
os @ react-dom.production.min.js:209
n.callback @ react-dom.production.min.js:226
vo @ react-dom.production.min.js:131
us @ react-dom.production.min.js:212
Ml @ react-dom.production.min.js:255
t.unstable_runWithPriority @ scheduler.production.min.js:19
Vi @ react-dom.production.min.js:122
wl @ react-dom.production.min.js:248
ll @ react-dom.production.min.js:239
(anonymous) @ react-dom.production.min.js:123
t.unstable_runWithPriority @ scheduler.production.min.js:19
Vi @ react-dom.production.min.js:122
Zi @ react-dom.production.min.js:123
Ki @ react-dom.production.min.js:122
rl @ react-dom.production.min.js:230
enqueueSetState @ react-dom.production.min.js:132
_.setState @ react.production.min.js:12
(anonymous) @ ExplicitSize.jsx:105
c @ throttle.js:32
(anonymous) @ ExplicitSize.jsx:64
Dimension.ts:1265 Uncaught TypeError: this._query.aggregations is not a function
    at n.value (Dimension.ts:1265:39)
    at n.value (Dimension.ts:1282:30)
    at fu (DimensionInfoPopover.tsx:37:49)
    at Qo (react-dom.production.min.js:153:146)
    at xs (react-dom.production.min.js:261:496)
    at yl (react-dom.production.min.js:246:265)
    at ml (react-dom.production.min.js:246:194)
    at ll (react-dom.production.min.js:239:172)
    at react-dom.production.min.js:123:115
    at t.unstable_runWithPriority (scheduler.production.min.js:19:467)
value @ Dimension.ts:1265
value @ Dimension.ts:1282
fu @ DimensionInfoPopover.tsx:37
Qo @ react-dom.production.min.js:153
xs @ react-dom.production.min.js:261
yl @ react-dom.production.min.js:246
ml @ react-dom.production.min.js:246
ll @ react-dom.production.min.js:239
(anonymous) @ react-dom.production.min.js:123
t.unstable_runWithPriority @ scheduler.production.min.js:19
Vi @ react-dom.production.min.js:122
Zi @ react-dom.production.min.js:123
Ki @ react-dom.production.min.js:122
rl @ react-dom.production.min.js:230
enqueueSetState @ react-dom.production.min.js:132
_.setState @ react.production.min.js:12
(anonymous) @ ExplicitSize.jsx:105
c @ throttle.js:32
(anonymous) @ ExplicitSize.jsx:64
StructuredQuery.ts:340 Warning: can't clean query without metadata!
value @ StructuredQuery.ts:340
value @ Question.ts:1210
value @ Question.ts:936
C @ urls.js:95
value @ Navbar.jsx:171
value @ Navbar.jsx:286
Wa @ react-dom.production.min.js:182
Ba @ react-dom.production.min.js:181
xs @ react-dom.production.min.js:263
yl @ react-dom.production.min.js:246
ml @ react-dom.production.min.js:246
ll @ react-dom.production.min.js:239
(anonymous) @ react-dom.production.min.js:123
t.unstable_runWithPriority @ scheduler.production.min.js:19
Vi @ react-dom.production.min.js:122
Zi @ react-dom.production.min.js:123
Ki @ react-dom.production.min.js:122
rl @ react-dom.production.min.js:230
enqueueSetState @ react-dom.production.min.js:132
_.setState @ react.production.min.js:12
l.onStateChange @ connectAdvanced.js:207
notify @ Subscription.js:23
t.notifyNestedSubs @ Subscription.js:62
l.onStateChange @ connectAdvanced.js:204
notify @ Subscription.js:23
t.notifyNestedSubs @ Subscription.js:62
l.onStateChange @ connectAdvanced.js:204
d @ createStore.js:172
(anonymous) @ middleware.js:22
(anonymous) @ index.js:28
(anonymous) @ store.js:25
dispatch @ applyMiddleware.js:35
(anonymous) @ redux.js:204
u @ runtime.js:63
(anonymous) @ runtime.js:293
(anonymous) @ runtime.js:118
v @ app-main.bundle.js?be1dafa68db102c8e473:2
a @ app-main.bundle.js?be1dafa68db102c8e473:2
Promise.then (async)
v @ app-main.bundle.js?be1dafa68db102c8e473:2
a @ app-main.bundle.js?be1dafa68db102c8e473:2
(anonymous) @ app-main.bundle.js?be1dafa68db102c8e473:2
(anonymous) @ app-main.bundle.js?be1dafa68db102c8e473:2
(anonymous) @ redux.js:200
(anonymous) @ store.js:23
(anonymous) @ bindActionCreators.js:3
(anonymous) @ EntityListLoader.jsx:155
u @ runtime.js:63
(anonymous) @ runtime.js:293
(anonymous) @ runtime.js:118
k @ app-main.bundle.js?be1dafa68db102c8e473:2
a @ app-main.bundle.js?be1dafa68db102c8e473:2
(anonymous) @ app-main.bundle.js?be1dafa68db102c8e473:2
(anonymous) @ app-main.bundle.js?be1dafa68db102c8e473:2
(anonymous) @ EntityListLoader.jsx:150
value @ EntityListLoader.jsx:171
us @ react-dom.production.min.js:212
Ml @ react-dom.production.min.js:255
t.unstable_runWithPriority @ scheduler.production.min.js:19
Vi @ react-dom.production.min.js:122
wl @ react-dom.production.min.js:248
ll @ react-dom.production.min.js:239
(anonymous) @ react-dom.production.min.js:123
t.unstable_runWithPriority @ scheduler.production.min.js:19
Vi @ react-dom.production.min.js:122
Zi @ react-dom.production.min.js:123
Ki @ react-dom.production.min.js:122
rl @ react-dom.production.min.js:230
enqueueSetState @ react-dom.production.min.js:132
_.setState @ react.production.min.js:12
(anonymous) @ ExplicitSize.jsx:105
c @ throttle.js:32
(anonymous) @ ExplicitSize.jsx:64
Show 42 more frames
StructuredQuery.ts:340 Warning: can't clean query without metadata!
value @ StructuredQuery.ts:340
value @ Question.ts:1210
value @ Question.ts:936
C @ urls.js:95
value @ Navbar.jsx:171
value @ Navbar.jsx:286
Wa @ react-dom.production.min.js:182
Ba @ react-dom.production.min.js:181
xs @ react-dom.production.min.js:263
yl @ react-dom.production.min.js:246
ml @ react-dom.production.min.js:246
ll @ react-dom.production.min.js:239
(anonymous) @ react-dom.production.min.js:123
t.unstable_runWithPriority @ scheduler.production.min.js:19
Vi @ react-dom.production.min.js:122
Zi @ react-dom.production.min.js:123
Ki @ react-dom.production.min.js:122
rl @ react-dom.production.min.js:230
enqueueSetState @ react-dom.production.min.js:132
_.setState @ react.production.min.js:12
l.onStateChange @ connectAdvanced.js:207
notify @ Subscription.js:23
t.notifyNestedSubs @ Subscription.js:62
l.onStateChange @ connectAdvanced.js:204
notify @ Subscription.js:23
t.notifyNestedSubs @ Subscription.js:62
l.onStateChange @ connectAdvanced.js:204
d @ createStore.js:172
(anonymous) @ middleware.js:22
(anonymous) @ index.js:28
(anonymous) @ store.js:25
dispatch @ applyMiddleware.js:35
action @ store.js:20
(anonymous) @ actions.js:631
u @ runtime.js:63
(anonymous) @ runtime.js:293
(anonymous) @ runtime.js:118
le @ app-main.bundle.js?be1dafa68db102c8e473:2
a @ app-main.bundle.js?be1dafa68db102c8e473:2
Promise.then (async)
le @ app-main.bundle.js?be1dafa68db102c8e473:2
a @ app-main.bundle.js?be1dafa68db102c8e473:2
Promise.then (async)
le @ app-main.bundle.js?be1dafa68db102c8e473:2
a @ app-main.bundle.js?be1dafa68db102c8e473:2
(anonymous) @ app-main.bundle.js?be1dafa68db102c8e473:2
(anonymous) @ app-main.bundle.js?be1dafa68db102c8e473:2
(anonymous) @ actions.js:370
(anonymous) @ store.js:23
(anonymous) @ bindActionCreators.js:3
value @ QueryBuilder.jsx:170
ko @ react-dom.production.min.js:137
Ba @ react-dom.production.min.js:176
xs @ react-dom.production.min.js:263
yl @ react-dom.production.min.js:246
ml @ react-dom.production.min.js:246
ll @ react-dom.production.min.js:239
(anonymous) @ react-dom.production.min.js:123
t.unstable_runWithPriority @ scheduler.production.min.js:19
Vi @ react-dom.production.min.js:122
Zi @ react-dom.production.min.js:123
Ki @ react-dom.production.min.js:122
rl @ react-dom.production.min.js:230
enqueueSetState @ react-dom.production.min.js:132
_.setState @ react.production.min.js:12
(anonymous) @ ExplicitSize.jsx:105
c @ throttle.js:32
(anonymous) @ ExplicitSize.jsx:64
Show 9 more frames
StructuredQuery.ts:340 Warning: can't clean query without metadata!
value @ StructuredQuery.ts:340
value @ Question.ts:1210
value @ Question.ts:936
C @ urls.js:95
value @ Navbar.jsx:171
value @ Navbar.jsx:286
Wa @ react-dom.production.min.js:182
Ba @ react-dom.production.min.js:181
xs @ react-dom.production.min.js:263
yl @ react-dom.production.min.js:246
ml @ react-dom.production.min.js:246
ll @ react-dom.production.min.js:239
(anonymous) @ react-dom.production.min.js:123
t.unstable_runWithPriority @ scheduler.production.min.js:19
Vi @ react-dom.production.min.js:122
Zi @ react-dom.production.min.js:123
Ki @ react-dom.production.min.js:122
rl @ react-dom.production.min.js:230
enqueueSetState @ react-dom.production.min.js:132
_.setState @ react.production.min.js:12
l.onStateChange @ connectAdvanced.js:207
notify @ Subscription.js:23
t.notifyNestedSubs @ Subscription.js:62
l.onStateChange @ connectAdvanced.js:204
d @ createStore.js:172
(anonymous) @ middleware.js:22
(anonymous) @ index.js:28
(anonymous) @ store.js:25
dispatch @ applyMiddleware.js:35
action @ store.js:20
(anonymous) @ actions.js:631
u @ runtime.js:63
(anonymous) @ runtime.js:293
(anonymous) @ runtime.js:118
le @ app-main.bundle.js?be1dafa68db102c8e473:2
a @ app-main.bundle.js?be1dafa68db102c8e473:2
Promise.then (async)
le @ app-main.bundle.js?be1dafa68db102c8e473:2
a @ app-main.bundle.js?be1dafa68db102c8e473:2
Promise.then (async)
le @ app-main.bundle.js?be1dafa68db102c8e473:2
a @ app-main.bundle.js?be1dafa68db102c8e473:2
(anonymous) @ app-main.bundle.js?be1dafa68db102c8e473:2
(anonymous) @ app-main.bundle.js?be1dafa68db102c8e473:2
(anonymous) @ actions.js:370
(anonymous) @ store.js:23
(anonymous) @ bindActionCreators.js:3
value @ QueryBuilder.jsx:170
ko @ react-dom.production.min.js:137
Ba @ react-dom.production.min.js:176
xs @ react-dom.production.min.js:263
yl @ react-dom.production.min.js:246
ml @ react-dom.production.min.js:246
ll @ react-dom.production.min.js:239
(anonymous) @ react-dom.production.min.js:123
t.unstable_runWithPriority @ scheduler.production.min.js:19
Vi @ react-dom.production.min.js:122
Zi @ react-dom.production.min.js:123
Ki @ react-dom.production.min.js:122
rl @ react-dom.production.min.js:230
enqueueSetState @ react-dom.production.min.js:132
_.setState @ react.production.min.js:12
(anonymous) @ ExplicitSize.jsx:105
c @ throttle.js:32
(anonymous) @ ExplicitSize.jsx:64
Show 6 more frames
NativeQueryEditor.jsx:346 Uncaught TypeError: Cannot read properties of null (reading 'offsetHeight')
    at i.value (NativeQueryEditor.jsx:346:29)
    at i.value (NativeQueryEditor.jsx:361:12)
    at o (debounce.js:13:29)
    at delay.js:7:17
value @ NativeQueryEditor.jsx:346
value @ NativeQueryEditor.jsx:361
o @ debounce.js:13
(anonymous) @ delay.js:7
setTimeout (async)
(anonymous) @ delay.js:6
(anonymous) @ restArguments.js:18
(anonymous) @ debounce.js:23
(anonymous) @ restArguments.js:16
ace.define.r._signal @ ace.js:3478
onChange @ ace.js:9382
ace.define.r._signal @ ace.js:3478
ace.define.applyDelta @ ace.js:7406
ace.define.insertMergedLines @ ace.js:7302
ace.define.insert @ ace.js:7230
ace.define.setValue @ ace.js:7136
setValue @ ace.js:12819
value @ NativeQueryEditor.jsx:250
value @ NativeQueryEditor.jsx:80
us @ react-dom.production.min.js:212
Ml @ react-dom.production.min.js:255
t.unstable_runWithPriority @ scheduler.production.min.js:19
Vi @ react-dom.production.min.js:122
wl @ react-dom.production.min.js:248
ll @ react-dom.production.min.js:239
(anonymous) @ react-dom.production.min.js:123
t.unstable_runWithPriority @ scheduler.production.min.js:19
Vi @ react-dom.production.min.js:122
Zi @ react-dom.production.min.js:123
Ki @ react-dom.production.min.js:122
rl @ react-dom.production.min.js:230
enqueueSetState @ react-dom.production.min.js:132
_.setState @ react.production.min.js:12
(anonymous) @ ExplicitSize.jsx:105
c @ throttle.js:32
(anonymous) @ ExplicitSize.jsx:64
4react-dom.production.min.js:49 Uncaught Error: Minified React error #188; visit https://reactjs.org/docs/error-decoder.html?invariant=188 for the full message or use the non-minified dev environment for full errors and additional helpful warnings.
    at react-dom.production.min.js:49:67
    at it (react-dom.production.min.js:50:217)
    at Object.t.findDOMNode (react-dom.production.min.js:289:336)
    at s.value (ExplicitSize.jsx:31:23)
    at ExplicitSize.jsx:101:28
    at c (throttle.js:32:21)
(anonymous) @ react-dom.production.min.js:49
it @ react-dom.production.min.js:50
t.findDOMNode @ react-dom.production.min.js:289
value @ ExplicitSize.jsx:31
(anonymous) @ ExplicitSize.jsx:101
c @ throttle.js:32
setTimeout (async)
value @ ExplicitSize.jsx:44
us @ react-dom.production.min.js:212
Ml @ react-dom.production.min.js:255
t.unstable_runWithPriority @ scheduler.production.min.js:19
Vi @ react-dom.production.min.js:122
wl @ react-dom.production.min.js:248
ll @ react-dom.production.min.js:239
(anonymous) @ react-dom.production.min.js:123
t.unstable_runWithPriority @ scheduler.production.min.js:19
Vi @ react-dom.production.min.js:122
Zi @ react-dom.production.min.js:123
Ki @ react-dom.production.min.js:122
rl @ react-dom.production.min.js:230
enqueueSetState @ react-dom.production.min.js:132
_.setState @ react.production.min.js:12
(anonymous) @ ExplicitSize.jsx:105
c @ throttle.js:32
(anonymous) @ ExplicitSize.jsx:64
react-dom.production.min.js:50 Uncaught Error: Minified React error #188; visit https://reactjs.org/docs/error-decoder.html?invariant=188 for the full message or use the non-minified dev environment for full errors and additional helpful warnings.
    at react-dom.production.min.js:50:151
    at it (react-dom.production.min.js:50:217)
    at Object.t.findDOMNode (react-dom.production.min.js:289:336)
    at s.value (ExplicitSize.jsx:31:23)
    at s.<anonymous> (ExplicitSize.jsx:101:28)
    at l (throttle.js:16:19)
```

</details>

**Information about your Metabase Installation:**
Tested 0.41.6 thru 0.42.4

--- a/frontend/src/metabase-lib/lib/Dimension.ts
+++ b/frontend/src/metabase-lib/lib/Dimension.ts
@@ -1472,7 +1472,11 @@ export class AggregationDimension extends Dimension {
    * Raw aggregation
    */
   _aggregation(): Aggregation {
-    return this._query && this._query.aggregations()[this.aggregationIndex()];
+    return (
+      this._query &&
+      this._query.aggregations &&
+      this._query.aggregations()[this.aggregationIndex()]
+    );
   }
 
   /**

--- a/frontend/test/metabase/scenarios/visualizations/reproductions/21615-convert-to-sql.cy.spec.js
+++ b/frontend/test/metabase/scenarios/visualizations/reproductions/21615-convert-to-sql.cy.spec.js
@@ -1,0 +1,35 @@
+import { restore, visitQuestionAdhoc } from "__support__/e2e/cypress";
+import { SAMPLE_DATABASE } from "__support__/e2e/cypress_sample_database";
+
+const { ORDERS, ORDERS_ID } = SAMPLE_DATABASE;
+
+describe("issue 21615", () => {
+  beforeEach(() => {
+    restore();
+    cy.signInAsAdmin();
+  });
+
+  it("should not throw an error when converting a table question to sql (metabase#21615)", () => {
+    visitQuestionAdhoc({
+      display: "table",
+      dataset_query: {
+        type: "query",
+        database: 1,
+        query: {
+          "source-table": ORDERS_ID,
+          aggregation: [["count"]],
+          breakout: [
+            ["field", ORDERS.CREATED_AT, { "temporal-unit": "month" }],
+          ],
+        },
+      },
+    });
+
+    cy.icon("notebook").click();
+    cy.icon("sql").click();
+    cy.button("Convert this question to SQL").click();
+
+    cy.findByText("Visualization").should("be.visible");
+    cy.findByText("Something went wrong").should("not.exist");
+  });
+});

